src/tails/node.ts

```diff
@@ -0,0 +1,229 @@
+import * as engine from "./engine";
+
+// Note: Dependencies are wonky here. For example, multiple vite plugins use
+// vite transparently. Next.js also uses React transitively. Should we "inherit"
+// things from vite in that case or just reimplement the few extra fields?
+// Maybe it's the builder that needs to know this deep inheritance?
+const allNodeFrameworks: engine.Framework[] = [
+  {
+    name: "core:express",
+    parent: "nodejs",
+    dependencies: [{ name: "express" }],
+  },
+  {
+    name: "core:nextjs",
+    parent: "nodejs",
+    requiredFiles: [["next.config.js", "next.config.ts"]],
+    dependencies: [{ name: "next" }],
+  },
+  {
+    name: "core:angular",
+    parent: "nodejs",
+    dependencies: [{ name: "@angular/core" }, { name: "@angular/cli" }],
+    buildCommand: "ng build",
+    devCommand: "ng run",
+  },
+  {
+    name: "core:astro",
+    parent: "nodejs",
+    requiredFiles: [["astro.config.mjs", "astro.config.cjs", "astro.config.js", "astro.config.ts"]],
+    dependencies: [{ name: "astrojs" }],
+    canEmbed: ["core:svelte", "core:react"],
```

Comment:
also core:svelte-vite, core:react-vite, since they might take a dep on a version of vite rather than leave it transitive